Bluejay/Bluejay/Bluejay.swift

```diff
@@ -123,6 +123,17 @@ public class Bluejay: NSObject {
         backgroundRestore restoreMode: BackgroundRestoreMode = .disable
         )
     {
+        /**
+         If a call to start is made while the app is still in the background (can happen if Bluejay is instantiated and started in the initialization of UIApplicationDelegate for example), Bluejay will mistake its unexpectedly early instantiation as an instantiation from background restoration.
+         
+         Therefore, an explicit call to start should assume that Bluejay is not initialized from background restoration, as the code flow for background restoration should not involve a call to start.
+        */
+        shouldRestoreState = false
```

Comment:
What do you think about checking `restoreMode` mode here? I think that an issue occurs with `restoreMode  == .disable`, since `start` does nothing here.

Reply:
If you instantiate Bluejay with `restoreMode == .disable` inside the `init` of UIApplicationDelegate, the problem of not being to connect isn't really caused by `case .disable: break`, it is caused by the `shouldRestoreState` getting stuck in `true` at the initialization of Bluejay.

So I think if we are now going to always sanitize `shouldRestoreState` by setting it back to false in cases of really early initialization of Bluejay, then leaving `case .disable: break` as is inside `start` should be fine.

Let me know if I didn't understand the question properly!

Reply:
Thank you for the explanation. I agree with you :)